libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -242,9 +259,21 @@ bool AP_Camera::update_location(const struct Location &loc)
         // be called without a new GPS fix
         return false;
     }
+
     if (get_distance(loc, _last_location) < _trigg_dist) {
         return false;
     }
-    _last_location = loc;
-    return true;
+
+    if ((unsigned) ahrs.roll_sensor/100.0f > _max_roll) {
```

Comment:
It should be fabsf()
On Nov 17, 2015 11:12 PM, "André Kjellstrup" <notifications@github.com>
wrote:

> In libraries/AP_Camera/AP_Camera.cpp
> <https://github.com/diydrones/ardupilot/pull/3194#discussion_r45166209>:
>
> >      if (get_distance(loc, _last_location) < _trigg_dist) {
> >          return false;
> >      }
> > -    _last_location = loc;
> > -    return true;
> > +
> > +    if ((unsigned) ahrs.roll_sensor/100.0f > _max_roll) {
>
> Thanks for the review.
> I did test actually this code, so I am not sure whatever you mean there is
> something wrong, or just redundant in it.
> . I wanted to prevent shooting of roll > x degrees.
> so by removing sign, I get 50 > 10 = true regardless of roll being 50 or
> -50. please tell how you think what line 267 should be.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/diydrones/ardupilot/pull/3194/files#r45166209>.
>